Equations.md

```diff
@@ -0,0 +1,289 @@
+SymPEP XXXX Equation class
+=========================================
+
+**Author** Jonathan Gutow, <gutow@uwosh.edu>
+
+**Status** Draft
+
+**Type** Standards Track
+
+**Created** 2021-01-25
+
+**Resolution** url to discussion (required for Accepted | Rejected | Withdrawn)
+
+
+## Abstract
+
+Fundamentally this is an object with two SymPy expressions connected by an equals sign.
+Potentially this could be extended to a general relational expression, but that is not
+part of this proposal. This class facilitates manipulating mathematical expressions
+in a manner as similar to what is done on paper as possible. For operations/methods where
+it makes sense they would be applied to both sides of the equation. As much as possible,
+operations on the equations should be initiated using syntax mimicking standard
+mathematical notation. This class also aims to provide output in standard mathematical
+notation when used within tools such as Jupyter, that support typeset LaTex.
+
+## Motivation and Scope
+The original impetus for development of the `Equation` class came from discussions J. Gutow 
+had with Physical Science Educators using SymPy in their classes. They wanted this on-paper-like
+behavior to facilitate demonstrating algebraic manipulations in physical science classes.
+
+There are a number of specific motivations for the development of the `Equation` class:
+1. Make interactive manipulation of equations as similar to doing the work on paper as
+possible and make the results easier to read by having results that look like
+`x = 2*y + 1` rather than `2*y + 1`, with what it is equal to on a separate line.
+1. Have equations on which manual algebra can be done by having SymPy perform the actual
+symbolic manipulation. This allows people to perform messy manipulations with fewer errors, much
+as calculators, spreadsheets and other numerical computation tools facilitate arithmetic.
+1. From an educational and new user perspective this increases the ease of use of SymPy
+by decreasing the cognitive load of using SymPy. This would be acheived by maintaining as
+close to standard mathematical notation for operations as possible (see 
+[Detailed Description](#detailed-description)). Making SymPy easier to begin using will
+increase its adoption and encourage people to learn to use symbolic math tools.
+1. The `Equality` class does not reliably support general SymPy operations on both sides
+of the equality without collapsing to a Boolean value of `True` or `False`. Parts of SymPy
+depend on this Boolean behavior; thus this proposal for a separate `Equation` class rather
+than an extension of `Equality`.
+1. Currently cycling between expressions and the results of the `solve` operation is quite
+messy. This proposal includes extending the `solve` operation so that if passed an equation
+or set of equations it would return the solutions as a set of equations.
+1. `.subs()` could also be made more natural to use if it could take a list of equations.
```

Comment:
This should not be confused with `xreplace` to be changed to unpack equation objects.
I think that for `xreplace`, the only `xreplace` to be allowed with equation is substituting the whole equation object with any other specified target, if equation matches some of the subtree.

Reply:
At this point I was only envisioning that `.subs(Eqn(a,b))` would be equivalent to `.subs({a:b})`. A wholesale update to `.subs()` should probably be a separate SymPEP or at least issue. Getting even more generic syntax such as `.subs(a=b)` being equivalent to `.subs({a:b})` will require more involved workarounds (preparsing?) as Sagemath uses. That is definitely beyond the scope of this proposal.

Reply:
It is possible to make Eqn recognized as some rule table (like how sympy's own Tuple or Dict are recognized as rule. But at least I'm sure that they shoul)
For making `subs(a=b)` work, I think that we can lookup for unknown keyword argument passed to `**kwargs` by `kwargs.items`, but the possible problem is conflicting with the options for `subs`, If the symbol is named like `simulatneous`.

Reply:
>For making s`ubs(a=b)` work, I think that we can lookup for unknown keyword argument passed to `**kwargs` by `kwargs.items`, but the possible problem is conflicting with the options for subs, If the symbol is named like `simulatneous`.

This is why I think `.subs(a=b)` would need more sophisticated preparsing to make it work.